daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/InterfacesTest.scala

```diff
@@ -130,56 +128,22 @@ class InterfacesTest
         )
       }
     }
-    "be unable to exercise an interface I4 choice via I3 on a T1 contract" in {
-      val command = ApiCommand.Exercise(idI3, cid2, "C4", ValueRecord(None, ImmArray.empty))
-      inside(runApi(command)) { case Left(Error.Interpretation(err, _)) =>
-        err shouldBe Error.Interpretation.DamlException(
-          IE.ContractDoesNotImplementRequiringInterface(idI3, idI4, cid2, idT2)
-        )
-      }
-    }
-
     "be able to exercise T1 by interface I1" in {
-      val command = ApiCommand.Exercise(idT1, cid1, "C1", ValueRecord(None, ImmArray.empty))
+      val command = ApiCommand.Exercise(idI1, cid1, "C1", ValueRecord(None, ImmArray.empty))
       runApi(command) shouldBe a[Right[_, _]]
     }
     "be able to exercise T2 by interface I1" in {
-      val command = ApiCommand.Exercise(idT2, cid2, "C1", ValueRecord(None, ImmArray.empty))
+      val command = ApiCommand.Exercise(idI1, cid2, "C1", ValueRecord(None, ImmArray.empty))
       runApi(command) shouldBe a[Right[_, _]]
     }
     "be able to exercise T2 by interface I2" in {
-      val command = ApiCommand.Exercise(idT2, cid2, "C2", ValueRecord(None, ImmArray.empty))
+      val command = ApiCommand.Exercise(idI2, cid2, "C2", ValueRecord(None, ImmArray.empty))
       runApi(command) shouldBe a[Right[_, _]]
     }
     "be unable to exercise T1 by interface I2 (stopped in preprocessor)" in {
       val command = ApiCommand.Exercise(idT1, cid1, "C2", ValueRecord(None, ImmArray.empty))
       preprocessApi(command) shouldBe a[Left[_, _]]
     }
```

Comment:
please also add a test where you try to exercise an interface choice (C1) by the template (T1) and it gets stopped in preprocessor; because you should only be able to exercise interface choice by interface id